### PR TITLE
feature/435: Download zip layer

### PR DESCRIPTION
**Pull request by zamuzakki**
This is PR for https://github.com/ConservationInternational/cplus-plugin/issues/435

### CODE REVIEW

File: django_project/cplus_api/models/layer.py [hunk 1]
```diff
@@ -131,6 +132,16 @@ def download_to_working_directory(self, base_dir):
                 destination.write(chunk)
         self.last_used_on = timezone.now()
         self.save(update_fields=['last_used_on'])
+        if file_path.endswith('.zip'):
+            with ZipFile(file_path, 'r') as zip_ref:
+                zip_ref.extractall(dir_path)
+            shapefile = [
+                file for file in os.listdir(dir_path) if file.endswith('.shp')
```
Comment (danangmassandy) [inline django_project/cplus_api/models/layer.py:139]: @zamuzakki can we find the correct shapefile by using the zip filename?
e.g. the zip is test_layer.zip, so the correct shp filename would be test_layer.shp
This is to ensure that this code returns correct shapefile when there are more than 1 extracted zip file.

File: django_project/cplus_api/utils/worker_analysis.py [hunk 2]
```diff
@@ -342,31 +346,37 @@ def create_and_upload_output_layer(
...
+            ),
+            shell=True
+        )
+    else:
+        final_output_path = file_path
     output_layer = OutputLayer.objects.create(
         name=filename,
         created_on=timezone.now(),
         owner=scenario_task.submitted_by,
         layer_type=BaseLayer.LayerTypes.RASTER,
```
Comment (danangmassandy) [inline django_project/cplus_api/utils/worker_analysis.py:372]: @zamuzakki is there possibility that the output is not raster? I see that you're using get_layer_type(file_path) in above lines, perhaps we can also change this line to use get_layer_type?

**Review by danangmassandy — CHANGES_REQUESTED** [review]

File: django_project/cplus_api/models/layer.py (hunk 1 above)
Comment (zamuzakki) [reply django_project/cplus_api/models/layer.py:139]: @danangmassandy That was what I did at first. But I was thinking about cases where the zip file name is different from the shapefile name. I'll update the extract location to a sub-directory, so each zip file would be extracted to different directory inside `dir_path`.

File: django_project/cplus_api/utils/worker_analysis.py (hunk 2 above)
Comment (zamuzakki) [reply django_project/cplus_api/utils/worker_analysis.py:372]: Yes @danangmassandy . When I added mask layer shapefile in settings, mask layer output is shapefile (not raster). I've updated it

```
{
    "final_output": [
        "/home/web/media/4/16f85857-836d-4469-8c3e-18ad6d587836/cplus_scenario_output_52e0.tif"
    ],
    "weighted_activities": [
        "/home/web/media/4/16f85857-836d-4469-8c3e-18ad6d587836/weighted_activities/Alien_Plant_Removal_1129_cleaned.tif",
        "/home/web/media/4/16f85857-836d-4469-8c3e-18ad6d587836/weighted_activities/Alien_Plant_Removal_ea73.tif"
    ],
    "normalized_pathways": [
        "/home/web/media/4/16f85857-836d-4469-8c3e-18ad6d587836/normalized_pathways/Alien_Plant_Removal_6995.tif",
        "/home/web/media/4/16f85857-836d-4469-8c3e-18ad6d587836/normalized_pathways/Avoided_Grassland_Conversion_27d7.tif"
    ],
    "normalized_activities": [
        "/home/web/media/4/16f85857-836d-4469-8c3e-18ad6d587836/normalized_activities/Alien_Plant_Removal_e25b.tif"
    ],
    "mask_layer": [
        "/home/web/media/4/16f85857-836d-4469-8c3e-18ad6d587836/mask_layer/mask.shp"
    ],
    "activities": [
        "/home/web/media/4/16f85857-836d-4469-8c3e-18ad6d587836/activities/Alien_Plant_Removal_0654.tif"
    ]
}
```

Comment (danangmassandy) [reply django_project/cplus_api/utils/worker_analysis.py]: @zamuzakki it seems the mask_layer is part of the input layer that we download before running the analysis. Please exclude this directory from the output.

Comment (zamuzakki) [reply django_project/cplus_api/utils/worker_analysis.py]: @danangmassandy I've updated it:
https://github.com/kartoza/cplus-api/pull/35/files#diff-465ccbd99fbb69a2027d5c32b4cbf690f75458225c5e715d594c9ae89645e69eR77

**Review by danangmassandy — APPROVED** [review]
Thanks Zakki!

**Pull request merged**